### PR TITLE
Password as CLI argument

### PR DESCRIPTION
**Pull request by Himura2la**
I don't want to enter password every time. I created run-scripts with the host and file and I need it to run quick.

### CODE REVIEW

**Comment by dpgeorge** [comment]
It would be preferable to specify the password using a switch, like `-p mypass` (and it's not hard to do this without argparse).  That's what users would expect.  Additional positional arguments would tend to indicate that you are copying multiple files.

Also, please don't use f-strings, they will only work with very recent versions of CPython.

**Comment by Himura2la** [comment]
Well, it's much harder to do WITHOUT `argparse` (which is not liked by maintainers). Multiple files are not supported so it's OK, one need a single glance on format to realize it.
I'll get rid of the interpolated string, OK.

**Comment by Himura2la** [comment]
I hope it's OK if `-p` argument must be the last one, trying to write another `argparse` will be rather complex task

**Comment by pfalcon** [comment]
> I hope it's OK if -p argument must be the last one, trying to write another argparse will be rather complex task

Better to let it appear anywhere. And no, you don't need to write another argparse for this (e.g. it's ok to assume -p and the actual password will be separated by a space).

**Comment by Himura2la** [comment]
IMHO adding so much abstract arguments syntax is [not a MicroPython way** either](https://github.com/micropython/webrepl/pull/29#issuecomment-307406421), but OK, I'll try to do it with not very many lines and changes...

**Comment by pfalcon** [comment]
There's definitely another alternative, which we followed so far: to treat a webrepl client implementation here as a simple, reference one and let community develop "value added" tools using it as an example.

But if there's desire to add a password arg to the reference client, it apparently should be done to not potentially confuse users (not in an adhoc way).

**Comment by Himura2la** [comment]
"value added" tools are not good, because passwords should never be hardcoded inside tools. Password should be near host address or nowhere.
Please check my last commit.

**Comment by Himura2la** [comment]
how about merge? *_*

File: webrepl_cli.py [hunk 1]
```diff
@@ -14,7 +14,7 @@
 # Treat this remote directory as a root for file transfers
 SANDBOX = ""
 #SANDBOX = "/tmp/webrepl/"
-DEBUG = 0
+OUTPUT_LEVEL = 1
```
Comment (dpgeorge) [inline webrepl_cli.py]: No need to change this.

File: webrepl_cli.py [hunk 2]
```diff
@@ -23,9 +23,12 @@
 
 
 def debugmsg(msg):
-    if DEBUG:
+    if OUTPUT_LEVEL > 1:
         print(msg)
 
+def infomsg(msg):
```
Comment (dpgeorge) [inline webrepl_cli.py]: No need to add this function.

File: webrepl_cli.py [hunk 3]
```diff
@@ -167,12 +170,12 @@ def help(rc=0):
     exename = sys.argv[0].rsplit("/", 1)[-1]
     print("%s - Perform remote file operations using MicroPython WebREPL protocol" % exename)
     print("Arguments:")
-    print("  <host>:<remote_file> <local_file> - Copy remote file to local file")
-    print("  <local_file> <host>:<remote_file> - Copy local file to remote file")
+    print("  <host>:<remote_file> <local_file> [-p password] - Copy remote file to local file")
+    print("  <local_file> <host>:<remote_file> [-p password] - Copy local file to remote file")
```
Comment (dpgeorge) [inline webrepl_cli.py]: It's customary to put options before the mandatory positional args.

File: webrepl_cli.py [hunk 4]
```diff
@@ -191,33 +194,43 @@ def parse_remote(remote):
...
         op = "get"
-        host, port, src_file = parse_remote(sys.argv[1])
-        dst_file = sys.argv[2]
+        host, port, src_file = parse_remote(arguments[1])
+        dst_file = arguments[2]
         if os.path.isdir(dst_file):
             basename = src_file.rsplit("/", 1)[-1]
             dst_file += "/" + basename
+        infomsg("%s %s from %s:%d (passwd:%s) and save as %s" % 
+                            (op, src_file, host, port, passwd, os.path.abspath(dst_file)))
```
Comment (dpgeorge) [inline webrepl_cli.py]: Please leave the original messages as they were.  It particular you don't want to print out the password!

File: webrepl_cli.py (hunk 2 above)
Comment (Himura2la) [reply webrepl_cli.py]: I'm afraid 
```python
if 1:
    print("op:%s, host:%s, port:%d, passwd:%s." % (op, host, port, passwd))
    print(src_file, "->", dst_file)
```
is a kind of bad code. But if you insist to leave it, OK then. Or maybe we could add a parameter to `debugmsg()` ?

**Comment by Himura2la** [comment]
OK, what do you think?

**Comment by Himura2la** [comment]
@dpgeorge ping

**Comment by bhollosi** [comment]
patch working as intended, this is a huge improvement to use webrepl in a shell script.
merge pls!

**Comment by Himura2la** [comment]
@dpgeorge @pfalcon ping

**Comment by pfalcon** [comment]
@Himura2la : Instead of pop'ing values out of sys.args, it would be better to just parse it once and assign values to local vars. Because your approach doesn't scale if more options are added.

But what really precluded your patch to be merged is that it contained unrelated changes. If you want to add a password option, the patch should contain the minimal changes to achieve it, and not a character more.

Anyway, thanks for persistence, I cleaned up the patch and merged it.

**Pull request closed without merging**